src/msw/textctrl.cpp

```diff
@@ -2718,14 +2725,53 @@ wxMenu *wxTextCtrl::MSWCreateContextMenu()
     return m;
 }
 
+namespace
+{
+
+inline void ApplyZoomFactor(HWND hwnd, float scale)
+{
+    UINT num = 1;
+    UINT denom = 1;
+    ::SendMessage(hwnd, EM_GETZOOM, (WPARAM)&num, (LPARAM)&denom);
+
+    float ratio = scale;
+    if (denom > 0)
+        ratio = ratio * (num / (float)denom);
+
+    num = 100 * ratio;
+    denom = 100;
```

Comment:
Why do we multiply by 100? It looks like multiplying by 12 or 24 or maybe 60 would be better as it would result in integer results for all the common scale factors.

Reply:
Windows seems to use 100 as default for denominator, at least when using ctrl+mousewheel to zoom. I added a comment in the code.